cachi2/core/package_managers/general.py

```diff
@@ -35,6 +47,98 @@ def download_binary_file(url, download_path, auth=None, insecure=False, chunk_si
             f.write(chunk)
 
 
+async def async_download_binary_file(
+    session: aiohttp_retry.RetryClient,
+    url: str,
+    download_path: Path,
+    auth: Optional[aiohttp.BasicAuth] = None,
+    chunk_size: int = 8192,
+) -> None:
+    """
+    Download a binary file (such as a TAR archive) from a URL using asyncio.
+
+    :param aiohttp_retry.RetryClient session: Aiohttp interface for making HTTP requests.
+    :param str url: URL for file download
+    :param str download_path: File path location
+    :param aiohttp.BasicAuth auth: Authentication for the URL
+    :param int chunk_size: Chunk size param for Response.content.read()
+    :raise FetchError: If download failed
+    """
+    try:
+        log.debug(f"Download started - {url}")
+        async with session.get(url, auth=auth, raise_for_status=True) as resp:
+            with open(download_path, "wb") as f:
+                while True:
+                    chunk = await resp.content.read(chunk_size)
+                    if not chunk:
+                        break
+                    f.write(chunk)
+
+    except Exception as exception:
+        log.error(f"Unsuccessful download: {url}")
+        # "from None" since we have the exception context in the logs
+        raise FetchError(
+            (f"exception_name: {exception.__class__.__name__}, " f"details: {exception}")
+        ) from None
+
+    log.debug(f"Download completed - {url}")
+
+
+async def async_download_files(
+    files_to_download: Dict[Any, Dict[str, Any]],  # dict[url, download path]
+    concurrency_limit: int,
+) -> None:
+    """Asynchronous function to download files.
+
+    :param files_to_download: Dict of files to download with file paths
+    :param concurrency_limit: Max number of concurrent tasks (downloads).
+    """
+
+    async def on_request_start(
+        session: Any,
+        trace_config_ctx: Any,
+        params: Any,
+    ) -> None:
```

Comment:
```suggestion
    async def on_request_start(
        session: aiohttp.ClientSession,
        trace_config_ctx: types.SimpleNamespace,
        params: aiohttp.TraceRequestStartParams,
    ) -> None:
```
(from the aiohttp types: [_SignalCallback](https://github.com/aio-libs/aiohttp/blob/a91a8aeebc9823ebd04c1516b3465922190e97e2/aiohttp/tracing.py#L18), [on_request_start](https://github.com/aio-libs/aiohttp/blob/a91a8aeebc9823ebd04c1516b3465922190e97e2/aiohttp/tracing.py#L131))

Reply:
Thanks!